server/tserver/src/main/java/org/apache/accumulo/tserver/logger/LogFileKey.java

```diff
@@ -189,4 +195,90 @@ public String toString() {
     }
     throw new RuntimeException("Unknown type of entry: " + event);
   }
+
+  /**
+   * Converts LogFileKey to Key. Creates a Key containing all of the LogFileKey fields. The fields
+   * are stored so the Key sorts maintaining the legacy sort order. The row of the Key is composed
+   * of 3 fields: EventNum + tabletID + seq (separated by underscore). The EventNum is the integer
+   * returned by eventType(). The column family is always the event. The column qualifier is
+   * dependent of the type of event and could be empty.
+   *
+   * <pre>
+   *     Key Schema:
+   *     Row = EventNum_tabletID_seq
+   *     Family = event
+   *     Qualifier = tserverSession OR filename OR KeyExtent
+   * </pre>
+   */
+  public Key toKey() throws IOException {
+    String row = "";
+    int eventNum = eventType(event);
+    String family = event.name();
+    String qual = "";
+    switch (event) {
+      case OPEN:
+        row = formatRow(eventNum, 0, 0);
+        qual = tserverSession;
+        break;
+      case COMPACTION_START:
+        row = formatRow(eventNum, tabletId, seq);
+        if (filename != null)
+          qual = filename;
+        break;
+      case MUTATION:
+      case MANY_MUTATIONS:
+      case COMPACTION_FINISH:
+        row = formatRow(eventNum, tabletId, seq);
+        break;
+      case DEFINE_TABLET:
+        row = formatRow(eventNum, tabletId, seq);
+        // Base64 encode KeyExtent
+        DataOutputBuffer buffer = new DataOutputBuffer();
+        tablet.writeTo(buffer);
+        qual = Base64.getEncoder().encodeToString(copyOf(buffer.getData(), buffer.getLength()));
+        buffer.close();
+        break;
+    }
+    return new Key(new Text(row), new Text(family), new Text(qual));
+  }
+
+  // format row = 1_000001_0000000001
+  private String formatRow(int eventNum, int tabletId, long seq) {
+    return String.format("%d_%06d_%010d", eventNum, tabletId, seq);
```

Comment:
Along time ago when writing continuous ingest, I found that String.format was super slow (It was the slowest thing in the data generation).  Not sure if that is still the case, may be something to consider and look into as this could have a lot of data going through it.

Reply:
Oh yeah I forgot about this. I will make it use `FastFormat`.

Reply:
See newer comments.